internal/pkg/kubernetes/client.go

```diff
@@ -0,0 +1,360 @@
+package kubernetes
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"io"
+	"octavius/internal/executor/config"
+	"octavius/internal/pkg/constant"
+	"os"
+	"path/filepath"
+	"time"
+
+	batch "k8s.io/api/batch/v1"
+	v1 "k8s.io/api/core/v1"
+	meta "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/watch"
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/rest"
+	"k8s.io/client-go/tools/clientcmd"
+)
+
+var (
+	typeMeta     meta.TypeMeta
+	ctx          context.Context
+	cancel       context.CancelFunc
+	timeoutError = errors.New("timeout when waiting job to be available")
+)
+
+func init() {
+	typeMeta = meta.TypeMeta{
+		Kind:       "Job",
+		APIVersion: "batch/v1",
+	}
+}
+
+type KubeClient interface {
+	ExecuteJob(jobId string, imageName string, envMap map[string]string) (string, error)
+	ExecuteJobWithCommand(jobId string, imageName string, args map[string]string, commands []string) (string, error)
+	JobExecutionStatus(executionName string) (string, error)
+	GetPodLogs(pod *v1.Pod) (io.ReadCloser, error)
+	WaitForReadyJob(executionName string, waitTime time.Duration) error
+	WaitForReadyPod(executionName string, waitTime time.Duration) (*v1.Pod, error)
+}
+
+type kubeClient struct {
+	clientSet                    kubernetes.Interface
+	namespace                    string
```

Comment:
why namespace is attached as property of kubeClient? if thats the case then this client can only be used just for a namespace?

Reply:
For the sake of MVP we are considering that an executor client will interact with only one kubernetes cluster. I guess we can modify this later when we move ahead with development